source/parsing/Token.cpp

```diff
@@ -370,6 +372,13 @@ Token Token::clone(BumpAllocator& alloc, span<Trivia const> trivia, string_view
     return result;
 }
 
+Token Token::deepClone(BumpAllocator& alloc) const {
+    SmallVectorSized<Trivia, 32> triviaBuffer;
+    for (const auto& t : trivia())
+        triviaBuffer.append(t.withLocation(alloc, location()));
```

Comment:
You should not use withLocation for this, since it results in different trivia than what we originally had here. You might instead need a deepClone method on the Trivia object as well, since a trivia might point to an allocation in the old allocator (like spans of tokens for example).